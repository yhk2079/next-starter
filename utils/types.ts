export type User = {
  id: string;
  name: string;
  email: string;
};

export type Users = User[];
