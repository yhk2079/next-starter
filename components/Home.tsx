import * as React from 'react';
import styled from 'styled-components';
import Users from './Users';

const Title = styled.div`
  display: flex;
  justify-content: center;
  color: ${({ theme }) => theme.colors.primary};
`;

const Home = React.memo(function Home() {
  return (
    <div>
      <Title>the world</Title>
      <Users />
    </div>
  );
});

export default Home;
