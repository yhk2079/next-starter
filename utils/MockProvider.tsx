import { QueryClientProvider, QueryClient } from 'react-query';

const queryClient = new QueryClient();
const MockProvider: React.FC = ({ children }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export default MockProvider;
