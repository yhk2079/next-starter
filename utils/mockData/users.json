[
  {
    "id": "1",
    "name": "test",
    "email": "test@email.com"
  }
]
