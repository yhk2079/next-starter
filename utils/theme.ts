export const theme = {
  colors: {
    primary: '#0070f3',
  },
};

export type Theme = typeof theme;

export default theme;
