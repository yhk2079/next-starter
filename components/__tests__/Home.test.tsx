import { render, waitFor } from '@testing-library/react';
import nock from 'nock';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../utils/theme';
import MockProvider from '../../utils/MockProvider';
import users from '../../utils/mockData/users.json';
import Home from '../Home';

test('should render Home with axios and nock', async () => {
  // TODO: compare msw and nock .
  nock('http://localhost:1080')
    .defaultReplyHeaders({
      'access-control-allow-origin': '*',
      'access-control-allow-credentials': 'true',
    })
    .get('/users')
    .reply(200, users);

  const { getByText, asFragment } = render(
    <MockProvider>
      <ThemeProvider theme={theme}>
        <Home />
      </ThemeProvider>
    </MockProvider>,
  );
  expect(getByText('the world')).toBeInTheDocument();

  await waitFor(() => expect(getByText(/test@email.com/)).toBeInTheDocument());
  expect(asFragment()).toMatchSnapshot();
});
