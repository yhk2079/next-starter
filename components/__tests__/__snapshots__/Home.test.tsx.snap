// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Home with axios and nock 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Home__Title-sc-16woefq-0 zWCpI"
    >
      the world
    </div>
    <div>
      Users
      <div>
        <div>
          email: test@email.com
        </div>
        <div>
          name: test
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
